src/data/repositories/DataSetD2Api.ts

```diff
@@ -33,40 +33,85 @@ export class DataSetD2Api {
-                                      displayName: true,
-                                  },
-                              }
-                            : {}),
+            return this.getDataSets(options, attributes, coreCompetencies, attributesToFilter);
+        });
+    }
+
+    getWithOrgUnits(options: D2DataSetOptions): FutureData<Paginated<DataSet>> {
+        return this.getBaseData().flatMap(({ attributes, coreCompetencies }) => {
+            return this.getDataSetsWithOrgUnits(options, attributes, coreCompetencies, [
+                attributes.createdByApp.id,
+            ]);
+        });
+    }
+
+    private getDataSets(
+        options: D2DataSetOptions,
+        attributes: D2Config["attributes"],
+        coreCompetencies: CoreCompetency[],
+        attributesToFilter: Id[]
+    ): FutureData<Paginated<DataSet>> {
+        return apiToFuture(
+            this.api.models.dataSets.get({
+                pageSize: options.paging.pageSize,
+                page: options.paging.page,
+                filter: {
+                    "attributeValues.attribute.id": { in: attributesToFilter },
+                    "attributeValues.value": options.filters.projectsIds
+                        ? { in: [...this.getOrThrow(options.filters.projectsIds)] }
+                        : { eq: "true" },
+                    identifiable: { token: options.filters.search },
+                    id: { in: options.filters.ids },
+                },
+                fields: dataSetFields,
+                order: `${options.sorting.field}:${options.sorting.order}`,
+            })
+        ).flatMap(d2Response => {
+            const projectIds = this.getProjectIds(d2Response.objects, attributes);
+            return this.getProjectsByIds(projectIds).map(projects => {
+                const dataSets = d2Response.objects.map(d2DataSet => {
+                    return this.buildDataSet(d2DataSet, coreCompetencies, projects, attributes);
+                });
+                return { ...d2Response.pager, data: dataSets };
+            });
+        });
+    }
+
+    private getDataSetsWithOrgUnits(
+        options: D2DataSetOptions,
+        attributes: D2Config["attributes"],
+        coreCompetencies: CoreCompetency[],
+        attributesToFilter: Id[]
+    ): FutureData<Paginated<DataSet>> {
+        return apiToFuture(
+            this.api.models.dataSets.get({
+                pageSize: options.paging.pageSize,
+                page: options.paging.page,
+                filter: {
+                    "attributeValues.attribute.id": { in: attributesToFilter },
+                    "attributeValues.value": options.filters.projectsIds
+                        ? { in: [...this.getOrThrow(options.filters.projectsIds)] }
+                        : { eq: "true" },
+                    identifiable: { token: options.filters.search },
+                    id: { in: options.filters.ids },
+                },
+                fields: {
```

Comment:
a lot of repeated code with the previous method, abstract (if it's conceptually the same)